src/Build/BackEnd/Components/BuildRequestEngine/BuildRequestEntry.cs

```diff
@@ -171,68 +144,33 @@ internal BuildRequestEntry(BuildRequest request, BuildRequestConfiguration reque
         /// <summary>
         /// Returns the object used to lock for synchronization of long-running operations.
         /// </summary>
-        public Object GlobalLock
-        {
-            [DebuggerStepThrough]
-            get
-            { return _dataMonitor; }
-        }
+        public Object GlobalLock { [DebuggerStepThrough] get; }
 
         /// <summary>
         /// Returns the root directory for the project being built by this request.
         /// </summary>
-        public string ProjectRootDirectory
-        {
-            get
-            {
-                if (_projectRootDirectory == null)
-                {
-                    _projectRootDirectory = Path.GetDirectoryName(RequestConfiguration.ProjectFullPath);
-                }
-
-                return _projectRootDirectory;
-            }
-        }
+        public string ProjectRootDirectory => _projectRootDirectory ??
+                                              (_projectRootDirectory = Path.GetDirectoryName(RequestConfiguration.ProjectFullPath));
 
         /// <summary>
         /// Returns the current state of the build request.
         /// </summary>
-        public BuildRequestEntryState State
-        {
-            [DebuggerStepThrough]
-            get
-            { return _state; }
-        }
+        public BuildRequestEntryState State { [DebuggerStepThrough] get; private set; }
```

Comment:
Does `[DebuggerStepThrough]` work for autoprops? I thought it would be unnecessary.

Reply:
Didn't find a strong opinion when looking around for this. I'll proceed to remove these from the auto props - some here and more in later PRs.